src/functions/Add-PuppetReferenceDocumentation.ps1

```diff
@@ -0,0 +1,40 @@
+function Add-PuppetReferenceDocumentation {
+    <#
+      .SYNOPSIS
+        Generate REFERENCE.md
+      .DESCRIPTION
+        Generate REFERENCE.md
+      .PARAMETER PuppetModuleFolderPath
+        The path, relative or literal, to the Puppet module's root folder.
+      .EXAMPLE
+        Add-PuppetReferenceDocumentation -PuppetModuleFolderPath C:\output\testmodule
+        This command will generate `REFERENCE.md` file.
+    #>
+    [CmdletBinding()]
+    param (
+      [Parameter(Mandatory=$true)]
+      $PuppetModuleFolderPath
+    )
+
+    begin {
+      $PuppetModuleFolderPath = Resolve-Path -Path $PuppetModuleFolderPath -ErrorAction Stop
+      $Command = [scriptblock]::Create("pdk bundle exec puppet strings generate --format markdown --out REFERENCE.md")
+    }
+    process {
+      Try {
+        $ErrorActionPreference = 'Stop'
+        Invoke-PdkCommand -Path $PuppetModuleFolderPath -Command $Command -SuccessFilterScript {
+          $_ -match "pdk"
+        }
+        # Verify REFERENCE.md file is generated
+        $ReferenceFile = Join-Path -Path $PuppetModuleFolderPath -ChildPath REFERENCE.md
+        if (Test-Path $ReferenceFile -PathType leaf)
+          {
+            Write-Output "REFERENCE.md file generated"
```

Comment:
Instead of `Write-Output` we may want to either write a verbose message (if we want to provide it as optional information) or invert the logic to raise an error here if the PDK command completed successfully but the file can't be found (is this state possible? not sure).

If we _do_ want to write a verbose log, we'll need to use [`Write-PSFMessage`](https://psframework.org/documentation/commands/PSFramework/Write-PSFMessage.html) from the PSFramework module, like:

```powershell
Write-PSFMessage "foo"
```